src/common/mfu_flist_copy.c

```diff
@@ -1554,50 +1583,76 @@ static int mfu_is_eof(const char* file, int fd)
 static int mfu_copy_file_normal(
     const char* src,
     const char* dest,
-    const int in_fd,
-    const int out_fd,
     uint64_t offset,
     uint64_t length,
     uint64_t file_size,
-    mfu_copy_opts_t* mfu_copy_opts)
+    mfu_copy_opts_t* mfu_copy_opts,
+    mfu_file_t* mfu_src_file,
+    mfu_file_t* mfu_dst_file)
 {
-    /* hint that we'll read from file sequentially */
-//    posix_fadvise(in_fd, offset, chunk_size, POSIX_FADV_SEQUENTIAL);
-
+    /* There is no equivalent "lseek" in daos, it is part of read, so skip that
+     * for a DAOS path */
     /* seek to offset in source file */
-    if(mfu_lseek(src, in_fd, offset, SEEK_SET) == (off_t)-1) {
-        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in source path `%s' (errno=%d %s)",
-            src, errno, strerror(errno));
+    if(mfu_file_lseek(src, mfu_src_file, offset, SEEK_SET) == (off_t)-1) {
+        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in source path `%s' (errno=%d %s)", src, errno, strerror(errno));
         return -1;
     }
 
     /* seek to offset in destination file */
-    if(mfu_lseek(dest, out_fd, offset, SEEK_SET) == (off_t)-1) {
-        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in destination path `%s' (errno=%d %s)",
-            dest, errno, strerror(errno));
+    if (mfu_file_lseek(dest, mfu_dst_file, offset, SEEK_SET) == (off_t)-1) {
+        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in destination path `%s' (errno=%d %s)", dest, errno, strerror(errno));
         return -1;
     }
 
-    /* get buffer */
+    /* declare daos vars */
+#ifdef DAOS_SUPPORT
+    daos_size_t got_size;
+    d_sg_list_t sgl;
+    d_iov_t     iov;
+#endif
+
+    /* set buffer and buffer size */
     size_t buf_size = mfu_copy_opts->block_size;
-    void* buf = mfu_copy_opts->block_buf1;
+    void* buf       = mfu_copy_opts->block_buf1;
+
+    /* set daos vars and buffers */
+#ifdef DAOS_SUPPORT
+    if (mfu_src_file->type == DAOS || mfu_dst_file->type == DAOS) {
+        sgl.sg_nr = 1;
+        D_ALLOC(buf, buf_size);
+        D_ASSERT(buf != NULL);
+        d_iov_set(&iov, buf, buf_size);
+        sgl.sg_iovs = &iov;
+        sgl.sg_nr_out = 1;
+        if (mfu_src_file->type == DAOS) {
+            mfu_src_file->offset = (daos_off_t) offset;
```

Comment:
If you update the daos_lseek to set the mfu_file->offset, then you can remove this line.